### PR TITLE
[DataPipe] Export ShardingFilter

### PR DESCRIPTION
**Pull request by ejguan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#69844 Export ShardingFilter**

Differential Revision: [D33062183](https://our.internmc.facebook.com/intern/diff/D33062183)

### CODE REVIEW

File: torch/utils/data/datapipes/iter/combining.py [hunk 1]
```diff
@@ -44,18 +44,6 @@ def __len__(self) -> int:
...
-class IterateBuffer(IterDataPipe):
-
-    def __init__(self, buffer):
-        self.buffer = buffer
-
-    def __iter__(self):
-        for i in self.buffer:
-            yield i
-
-
```
Comment (ejguan) [inline torch/utils/data/datapipes/iter/combining.py:58]: Removing it because it's same as `IterableWrapper`

**Review by NivekT — APPROVED** [review]
LGTM. Thank you!

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33062183).

**Comment by nateanl** [comment]
Hi @ejguan, I met an import error though I installed the nightly torch. Here is my python script: 
```python
from torch.utils.data.datapipes.iter import ShardingFilter
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'ShardingFilter' from 'torch.utils.data.datapipes.iter' (/data/home/miniconda3/envs/torchdata/lib/python3.8/site-packages/torch/utils/data/datapipes/iter/__init__.py)
```
Is it expected, or should I wait until the nightly build includes the change?

**Comment by ejguan** [comment]
@nateanl 
It's expected, but I didn't expect PyTorch Core has been struggled 3 days to update nightly. Phillip has a PR to temporarily fix it.

**Comment by nateanl** [comment]
By updating the pytorch via
```
pip install --pre torch torchvision torchtext torchaudio -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html --upgrade
```
I still met the same error. Perhaps there's something wrong in the nightly build...

**Comment by ejguan** [comment]
> By updating the pytorch via
> 
> ```
> pip install --pre torch torchvision torchtext torchaudio -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html --upgrade
> ```
> 
> I still met the same error. Perhaps there's something wrong in the nightly build...

The PR in PyTorch Core has not got into nightly build yet.

**Comment by nateanl** [comment]
I see. Thanks for letting me know!

**Pull request closed without merging**